### PR TITLE
[bat-native-ledger] Cleanup usage of base::Optional

### PR DESCRIPTION
**Pull request by mariospr**
This type is deprecated and we should be using absl::optional now and
avoiding the old one in newly added code, so that we don't have to keep
changing that when rebasing the cr93 branch.

This CL removes the few old types remaining in master after the main
cleanup done at https://github.com/brave/brave-core/commit/1c989344.

Resolves https://github.com/brave/brave-browser/issues/16589

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

N/A

### CODE REVIEW

**Review by bsclifton — APPROVED** [review]
LGTM; let's get another review before merge though 😄

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.cc
Comment (zenparsing) [inline vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.cc:16]: Do we need these includes (or if not, are they desirable)? The JSON reader/writer headers would bring it in.

**Review by zenparsing — APPROVED** [review]

File: vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.cc
Comment (mariospr) [reply vendor/bat-native-ledger/src/bat/ledger/internal/endpoint/gemini/post_account/post_account_gemini.cc:16]: I think it's expected to include them according to the IWYU rule that's in effect for Chromium:
https://google.github.io/styleguide/cppguide.html#Include_What_You_Use

**Review by szilardszaloki — APPROVED** [review]

**Review by jumde — APPROVED** [review]

**Pull request merged**
